diff -Nur inn-2.4.0.orig/authprogs/ckpasswd.c inn-2.4.0/authprogs/ckpasswd.c
--- inn-2.4.0.orig/authprogs/ckpasswd.c	2003-05-09 06:25:27.000000000 +0200
+++ inn-2.4.0/authprogs/ckpasswd.c	2003-06-29 13:03:12.000000000 +0200
@@ -26,13 +26,8 @@
 #include <grp.h>
 
 #if HAVE_DBM
-# if HAVE_NDBM_H
-#  include <ndbm.h>
-# elif HAVE_GDBM_NDBM_H
-#  include <gdbm-ndbm.h>
-# elif HAVE_DB1_NDBM_H
-#  include <db1/ndbm.h>
-# endif
+# define DB_DBM_HSEARCH 1
+# include <db.h>
 # define OPT_DBM "d:"
 #else
 # define OPT_DBM ""
--- inn-2.4.2/configure.in.orig	2004-12-22 05:21:19.000000000 +0100
+++ inn-2.4.2/configure.in	2004-12-27 08:32:38.010338472 +0100
@@ -780,7 +780,7 @@
 dnl Search for various additional libraries used by portions of INN.
 INN_SEARCH_AUX_LIBS(crypt, crypt, CRYPT_LIB)
 INN_SEARCH_AUX_LIBS(getspnam, shadow, SHADOW_LIB)
-INN_SEARCH_AUX_LIBS(dbm_open, ndbm dbm, DBM_LIB,
+INN_SEARCH_AUX_LIBS(db_create, db-4.6 db-4.5 db-4.4 db-4.3 db-4.2 db-4.1 db4 db, DBM_LIB,
     [AC_DEFINE(HAVE_DBM, 1, [Define if you have a dbm library.])])
 
 dnl IRIX has a PAM library with the right symbols but no header files suitable
@@ -973,17 +973,11 @@
 AC_HEADER_SYS_WAIT
 
 dnl Generic checks for header files.
-AC_CHECK_HEADERS(crypt.h inttypes.h limits.h ndbm.h pam/pam_appl.h stdbool.h \
+AC_CHECK_HEADERS(crypt.h inttypes.h limits.h pam/pam_appl.h stdbool.h \
                  stddef.h stdint.h string.h sys/bitypes.h sys/filio.h \
                  sys/loadavg.h sys/param.h sys/select.h sys/sysinfo.h \
                  sys/time.h unistd.h)
 
-dnl Some Linux systems have db1/ndbm.h instead of ndbm.h.  Others have
-dnl gdbm-ndbm.h.
-if test x"$ac_cv_header_ndbm_h" = xno ; then
-    AC_CHECK_HEADERS(db1/ndbm.h gdbm-ndbm.h)
-fi
-
 dnl Check to see if herrno is declared.
 AC_DEFUN([INN_NEED_HERRNO_DECLARATION],
 [AC_CACHE_CHECK([whether h_errno must be declared], inn_cv_herrno_need_decl,
