--- inn-2.4.4/authprogs/ckpasswd.c.orig	2008-05-05 12:18:36.000000000 +0200
+++ inn-2.4.4/authprogs/ckpasswd.c	2008-06-12 23:17:16.157845226 +0200
@@ -26,11 +26,11 @@
 #include <grp.h>
 
 #if defined(HAVE_DBM) || defined(HAVE_BDB_DBM)
-# if HAVE_NDBM_H
-#  include <ndbm.h>
-# elif HAVE_BDB_DBM
+# if HAVE_BDB_DBM
 #  define DB_DBM_HSEARCH 1
 #  include <db.h>
+# elif HAVE_NDBM_H
+#  include <ndbm.h>
 # elif HAVE_GDBM_NDBM_H
 #  include <gdbm-ndbm.h>
 # elif HAVE_DB1_NDBM_H
--- inn-2.4.4/configure.in.orig	2008-05-05 12:18:36.000000000 +0200
+++ inn-2.4.4/configure.in	2008-06-12 23:24:11.005485208 +0200
@@ -670,8 +670,13 @@
             fi
         fi
     else
+	if test "$BERKELEY_DB_DIR" = "/usr" ; then
+        BERKELEY_DB_LDFLAGS=
+        BERKELEY_DB_CFLAGS=
+	else
         BERKELEY_DB_LDFLAGS="-L$BERKELEY_DB_DIR/lib"
         BERKELEY_DB_CFLAGS="-I$BERKELEY_DB_DIR/include"
+	fi
         BERKELEY_DB_LIB="-ldb"
         AC_MSG_RESULT($BERKELEY_DB_DIR)
     fi
@@ -993,17 +998,11 @@
 AC_HEADER_SYS_WAIT
 
 dnl Generic checks for header files.
-AC_CHECK_HEADERS(crypt.h inttypes.h limits.h ndbm.h pam/pam_appl.h stdbool.h \
+AC_CHECK_HEADERS(crypt.h inttypes.h limits.h pam/pam_appl.h stdbool.h \
                  stddef.h stdint.h string.h sys/bitypes.h sys/filio.h \
                  sys/loadavg.h sys/param.h sys/select.h sys/sysinfo.h \
                  sys/time.h unistd.h)
 
-dnl Some Linux systems have db1/ndbm.h instead of ndbm.h.  Others have
-dnl gdbm-ndbm.h.
-if test x"$ac_cv_header_ndbm_h" = xno ; then
-    AC_CHECK_HEADERS(db1/ndbm.h gdbm-ndbm.h)
-fi
-
 dnl Check to see if herrno is declared.
 AC_DEFUN([INN_NEED_HERRNO_DECLARATION],
 [AC_CACHE_CHECK([whether h_errno must be declared], inn_cv_herrno_need_decl,
