diff -Nur inn-2.4.0.orig/configure.in inn-2.4.0/configure.in
--- inn-2.4.0.orig/configure.in	2003-06-29 13:20:49.000000000 +0200
+++ inn-2.4.0/configure.in	2003-06-29 13:22:13.000000000 +0200
@@ -803,7 +803,7 @@
 AC_SEARCH_LIBS(setproctitle, util,
     [AC_DEFINE(HAVE_SETPROCTITLE, 1,
         [Define if you have the setproctitle function.])],
-    [LIBOBJS="$LIBOBJS setproctitle.o"
+    [AC_LIBOBJ([setproctitle])
      AC_CHECK_FUNCS(pstat)])
 
 dnl The rat's nest of networking libraries.  The common cases are not to
@@ -1406,7 +1406,7 @@
     AC_DEFINE([HAVE_INET_NTOA], 1,
               [Define if your system has a working inet_ntoa function.])
 else
-    LIBOBJS="$LIBOBJS inet_ntoa.${ac_objext}"
+    AC_LIBOBJ([inet_ntoa])
 fi])
 INN_FUNC_INET_NTOA
 
@@ -1554,7 +1554,7 @@
     AC_DEFINE([HAVE_SNPRINTF], 1,
         [Define if your system has a working snprintf function.])
 else
-    LIBOBJS="$LIBOBJS snprintf.${ac_objext}"
+    AC_LIBOBJ([snprintf])
 fi])
 INN_FUNC_SNPRINTF
 
