diff -Nur inn-2.4.0.orig/doc/man/rnews.1 inn-2.4.0/doc/man/rnews.1
--- inn-2.4.0.orig/doc/man/rnews.1	2003-05-09 06:25:27.000000000 +0200
+++ inn-2.4.0/doc/man/rnews.1	2003-06-29 12:58:27.000000000 +0200
@@ -67,7 +67,7 @@
 is defined and the command is any other word, then
 .I rnews
 will try to execute a program with that name in the directory
-.IR <pathbin\ in\ inn.conf>/bin/rnews.libexec .
+.IR /usr/lib/news/rnews .
 
 The batch will be fed into the program's standard input, and the
 standard output will be read back as input into
diff -Nur inn-2.4.0.orig/include/paths.h.in inn-2.4.0/include/paths.h.in
--- inn-2.4.0.orig/include/paths.h.in	2003-05-09 06:25:27.000000000 +0200
+++ inn-2.4.0/include/paths.h.in	2003-06-29 12:57:56.000000000 +0200
@@ -19,12 +19,12 @@
 /* Default prefix path is pathbin. */
 #define _PATH_NNRPD		"nnrpd"
 #define _PATH_NNTPD		"nnrpd"
-#define _PATH_AUTHDIR		"auth"
+#define _PATH_AUTHDIR		"/usr/lib/news/auth"
 #define _PATH_AUTHDIR_GENERIC	"generic"
 #define _PATH_AUTHDIR_NOPASS	"resolv"
 #define _PATH_AUTHDIR_PASSWD	"passwd"
 #define _PATH_CTLINND		"ctlinnd"
-#define _PATH_RNEWSPROGS	"rnews.libexec"
+#define _PATH_RNEWSPROGS	"/usr/lib/news/rnews"
 
 /* Default prefix path is pathfilter. */
 #define _PATH_TCL_STARTUP	"startup.tcl"
diff -Nur inn-2.4.0.orig/Makefile inn-2.4.0/Makefile
--- inn-2.4.0.orig/Makefile	2003-05-09 06:25:27.000000000 +0200
+++ inn-2.4.0/Makefile	2003-06-29 12:57:10.000000000 +0200
@@ -75,7 +75,7 @@
 ##  Installation rules.  make install installs everything; make update only
 ##  updates the binaries, scripts, and documentation and leaves config
 ##  files alone.
-install: directories
+install:
 	@for D in $(ALLDIRS) ; do \
 	    echo '' ; \
 	    cd $$D && $(MAKE) install || exit 1 ; cd .. ; \
@@ -91,9 +91,9 @@
 directories:
 	@chmod +x support/install-sh
 	for D in $(INSTDIRS) ; do \
-	    support/install-sh $(OWNER) -m 0755 -d $(D)$$D ; \
+	    support/install-sh -m 0755 -d $(D)$$D ; \
 	done
-	support/install-sh $(OWNER) -m 0750 -d $(D)$(PATHRUN)
+	support/install-sh -m 0750 -d $(D)$(PATHRUN)
 
 update: 
 	@chmod +x support/install-sh
--- inn-2.4.1.orig/Makefile.global.in	2003-05-09 06:25:27.000000000 +0200
+++ inn-2.4.1/Makefile.global.in	2003-06-29 12:59:13.000000000 +0200
@@ -234,16 +234,7 @@
 ##  configure; inews and rnews are special and have configure flags to
 ##  control how they're installed.  See INSTALL for more information.
 
-NEWSUSER        = @NEWSUSER@
-NEWSGROUP       = @NEWSGRP@
-
-INEWSMODE       = @INEWSMODE@
-RNEWSMODE       = @RNEWSMODE@
 FILEMODE        = @FILEMODE@
-
-OWNER           = -o $(NEWSUSER) -g $(NEWSGROUP)
-ROWNER          = -o $(NEWSUSER) -g @RNEWSGRP@
-
 INSTALL         = $(top)/support/install-sh -c
 
 ##  Installation commands.  These commands are used by the installation
diff -Nur inn-2.4.0.orig/scripts/scanlogs.in inn-2.4.0/scripts/scanlogs.in
--- inn-2.4.0.orig/scripts/scanlogs.in	2003-05-09 06:25:27.000000000 +0200
+++ inn-2.4.0/scripts/scanlogs.in	2003-06-29 12:58:56.000000000 +0200
@@ -11,7 +11,7 @@
 ##  then get a better egrep, like the FSF one.
 
 ##  Directory where old log files are kept.
-OLD=${MOST_LOGS}/OLD
+OLD=/var/log/archiv/news
 ##  If you want to archive the active file, enable this line.
 ACTIVEFILE=${ACTIVE}
 ##  Number of lines of error in each category to report.
