--- inn-2.2.1/backends/inndf.c.frsize	Tue Jul  6 09:38:41 1999
+++ inn-2.2.1/backends/inndf.c	Sun Aug 29 14:35:05 1999
@@ -80,7 +80,7 @@
 #define STATFUNCT	statvfs		/* function call */
 #define STATSTRUC	statvfs		/* structure name */
 #define STATAVAIL	f_bavail	/* blocks available */
-#define STATMULTI	f_frsize	/* fragment size/block size */
+#define STATMULTI	f_bsize		/* fragment size/block size */
 #define CAREFULL_STATMULTI f_bsize /* in case f_frsize is 0 */
 #define STATINODE	f_favail	/* inodes available */
 #define STATTYPES	u_long		/* type of f_bavail etc */
