diff -dur inn-2.4.1.orig/innd/chan.c inn-2.4.1/innd/chan.c
--- inn-2.4.1.orig/innd/chan.c	2004-02-09 10:40:15.000000000 +0100
+++ inn-2.4.1/innd/chan.c	2004-02-09 10:43:43.045917856 +0100
@@ -637,7 +637,7 @@
 int
 CHANreadtext(CHANNEL *cp)
 {
-    int	                i, j;
+    size_t              i, j;
     struct buffer       *bp;
     char		*p;
     int			oerrno;
@@ -915,7 +915,7 @@
 {
     static char		EXITING[] = "INND exiting because of signal\n";
     static int		fd;
-    int			i, j;
+    size_t		i, j;
     int			startpoint;
     int			count;
     int			lastfd;
Only in inn-2.4.1/innd: chan.c~
diff -dur inn-2.4.1.orig/innd/innd.h inn-2.4.1/innd/innd.h
--- inn-2.4.1.orig/innd/innd.h	2004-01-07 23:47:19.000000000 +0100
+++ inn-2.4.1/innd/innd.h	2004-02-09 10:45:24.224536368 +0100
@@ -358,9 +358,9 @@
   int		       HoldTime;
   time_t	       ArtBeg;
   int		       ArtMax;
-  int		       Start;		/* where current cmd/article starts
+  size_t	       Start;		/* where current cmd/article starts
 					   it indicates offset from bp->Data */
-  int		       Next;		/* next pointer to read
+  size_t	       Next;		/* next pointer to read
 					   it indicates offset from bp->Data */
   char		       Error[SMBUF];	/* error buffer */
   ARTDATA	       Data;		/* used for processing article */
Only in inn-2.4.1/innd: innd.h~
Only in inn-2.4.1/innd: nc.c~
