--- inn-2.4.4/authprogs/ckpasswd.c.orig	2008-05-05 12:18:36.000000000 +0200
+++ inn-2.4.4/authprogs/ckpasswd.c	2008-06-12 23:17:16.157845226 +0200
@@ -26,11 +26,11 @@
 #include <grp.h>
 
 #if defined(HAVE_DBM) || defined(HAVE_BDB_DBM)
-# if HAVE_NDBM_H
-#  include <ndbm.h>
-# elif HAVE_BDB_DBM
+# if HAVE_BDB_DBM
 #  define DB_DBM_HSEARCH 1
 #  include <db.h>
+# elif HAVE_NDBM_H
+#  include <ndbm.h>
 # elif HAVE_GDBM_NDBM_H
 #  include <gdbm-ndbm.h>
 # elif HAVE_DB1_NDBM_H
--- inn-2.5.3/m4/berkeleydb.m4.orig	2013-06-11 21:16:28.320439059 +0200
+++ inn-2.5.3/m4/berkeleydb.m4	2013-06-11 20:36:38.617155533 +0200
@@ -33,8 +33,13 @@
     if test x"$DB_DIR" = xyes ; then
         AC_MSG_ERROR([cannot find Berkeley DB])
     else
+	if test x"$DB_DIR" = x"/usr" ; then
+	DB_CPPFLAGS=
+	DB_LDFLAGS=
+	else
         DB_CPPFLAGS="-I$DB_DIR/include"
         DB_LDFLAGS="-L$DB_DIR/lib"
+	fi
         DB_LIBS="-ldb"
         AC_MSG_RESULT([$DB_DIR])
     fi
--- inn-2.5.3/configure.ac.orig	2013-06-11 21:17:03.440438325 +0200
+++ inn-2.5.3/configure.ac	2013-06-11 21:47:46.747066572 +0200
@@ -384,10 +384,6 @@
                   stdint.h string.h sys/bitypes.h sys/filio.h sys/loadavg.h \
                   sys/select.h sys/time.h unistd.h])
 
-dnl Some Linux systems have db1/ndbm.h instead of ndbm.h.  Others have
-dnl gdbm-ndbm.h.
-AC_CHECK_HEADERS([ndbm.h db1/ndbm.h gdbm-ndbm.h], [break])
-
 dnl Check for whether various symbols are declared.
 AC_CHECK_DECLS([pread, pwrite, snprintf, setproctitle, strlcat,
                 strlcpy, vsnprintf])
