diff -urNp -x '*.orig' inn-2.6.4.org/Makefile.global.in inn-2.6.4/Makefile.global.in
--- inn-2.6.4.org/Makefile.global.in	2021-09-02 23:41:49.119470657 +0200
+++ inn-2.6.4/Makefile.global.in	2021-09-02 23:41:49.199469335 +0200
@@ -344,7 +344,7 @@ INSTALL		= $(top)/support/install-sh -p
 
 BACKUP_OPTION	= -S .OLD
 
-LI_SPRI		= $(LIBTOOLINST) $(INSTALL) -o root -g $(RUNASGROUP) -m 4550 $(BACKUP_OPTION)
+LI_SPRI		= $(LIBTOOLINST) $(INSTALL) -m 4550 $(BACKUP_OPTION)
 LI_XPRI		= $(LIBTOOLINST) $(INSTALL) $(OWNER) -m 0550 $(BACKUP_OPTION)
 LI_XPUB		= $(LIBTOOLINST) $(INSTALL) $(OWNER) -m 0555 $(BACKUP_OPTION)
 LI_LPUB		= $(LIBTOOLINST) $(INSTALL) $(OWNER) -m 0555
