--- inn-2.5.3/doc/man/rnews.1.orig	2013-06-11 17:19:04.427402611 +0200
+++ inn-2.5.3/doc/man/rnews.1	2013-06-11 17:41:12.624041596 +0200
@@ -181,7 +181,7 @@
 .PP
 Otherwise, if the command is any other word, then \fBrnews\fR will try to
 execute a program with that name, looking for it in the directory
-\&\fIpathbin\fR/rnews.libexec.  The rest of the batch will be fed to that
+/usr/lib*/news/rnews .  The rest of the batch will be fed to that
 program's standard input, and the standard output from the program will be
 treated as if it were the original input to \fBrnews\fR.  \s-1INN\s0 comes with three
 such standard batch processors:
--- inn-2.5.3/include/inn/paths.h.in.orig	2013-06-11 17:42:12.677373675 +0200
+++ inn-2.5.3/include/inn/paths.h.in	2013-06-11 17:49:38.687364380 +0200
@@ -22,12 +22,12 @@
 /* Default prefix path is pathbin. */
 #define INN_PATH_NNRPD                  "nnrpd"
 #define INN_PATH_NNTPD                  "nnrpd"
-#define INN_PATH_AUTHDIR                "auth"
+#define INN_PATH_AUTHDIR                "@libdir@/news/auth"
 #define INN_PATH_AUTHDIR_GENERIC        "generic"
 #define INN_PATH_AUTHDIR_NOPASS         "resolv"
 #define INN_PATH_AUTHDIR_PASSWD         "passwd"
 #define INN_PATH_CTLINND                "ctlinnd"
-#define INN_PATH_RNEWSPROGS             "rnews.libexec"
+#define INN_PATH_RNEWSPROGS             "@libdir@/news/rnews"
 
 /* Default prefix path is pathfilter. */
 #define INN_PATH_PERL_STARTUP_INND      "startup_innd.pl"
--- inn-2.5.3/Makefile.global.in.orig	2013-06-11 17:54:53.957357808 +0200
+++ inn-2.5.3/Makefile.global.in	2013-06-11 17:57:11.437354942 +0200
@@ -250,7 +250,7 @@
 PATHAUTH	= $(PATHBIN)/auth
 PATHAUTHRESOLV	= $(PATHAUTH)/resolv
 PATHAUTHPASSWD	= $(PATHAUTH)/passwd
-PATHRNEWS	= $(PATHBIN)/rnews.libexec
+PATHRNEWS	= @libdir@/news/rnews
 PATHARCHIVE	= $(PATHSPOOL)/archive
 PATHARTICLES	= $(PATHSPOOL)/articles
 PATHINCOMING	= $(PATHSPOOL)/incoming
@@ -282,8 +282,8 @@
 RNEWSMODE	= @RNEWSMODE@
 FILEMODE	= @FILEMODE@
 
-OWNER		= -o $(RUNASUSER) -g $(RUNASGROUP)
-ROWNER		= -o $(RUNASUSER) -g $(RNEWSGROUP)
+OWNER		=
+ROWNER		=
 
 INSTALL		= $(top)/support/install-sh -c
 
--- inn-2.5.3/scripts/scanlogs.in.orig	2013-06-11 17:57:46.774020873 +0200
+++ inn-2.5.3/scripts/scanlogs.in	2013-06-11 17:59:15.287352361 +0200
@@ -10,7 +10,7 @@
 ##	norotate	Do not rotate logfiles.
 
 ##  Directory where old log files are kept.
-OLD=${MOST_LOGS}/OLD
+OLD=/var/log/archive/news
 
 ##  Files defined in innshellvars.  We repeat them for clarity.
 ERRLOG=${MOST_LOGS}/errlog
--- inn-2.5.3/scripts/inncheck.in.orig	2013-06-11 17:59:47.974018346 +0200
+++ inn-2.5.3/scripts/inncheck.in	2013-06-11 18:00:16.837351078 +0200
@@ -44,7 +44,7 @@
     'passwd.nntp',	"$INN::Config::pathetc/passwd.nntp",
     'readers.conf',	"$INN::Config::pathetc/readers.conf",
     'rnews',		"$INN::Config::rnews",
-    'rnewsprogs',	"$INN::Config::pathbin/rnews.libexec",
+    'rnewsprogs',	"@libdir@/news/rnews",
     'spooltemp',	"$INN::Config::pathtmp",
     'spool',		"$INN::Config::patharticles",
     'spoolnews',	"$INN::Config::pathincoming",
--- inn-2.5.3/frontends/Makefile.orig	2013-06-11 17:59:47.977351680 +0200
+++ inn-2.5.3/frontends/Makefile	2013-06-11 18:01:39.537349354 +0200
@@ -36,11 +36,11 @@
 	for F in getlist innconfval sm ; do \
 	    $(LI_XPUB) $$F $D$(PATHBIN)/$$F ; \
 	done
-	$(CP_XPUB) bunbatch $D$(PATHBIN)/rnews.libexec/bunbatch
-	$(CP_XPUB) c7unbatch $D$(PATHBIN)/rnews.libexec/c7unbatch
-	$(LI_XPUB) decode $D$(PATHBIN)/rnews.libexec/decode
-	$(LI_XPUB) encode $D$(PATHBIN)/rnews.libexec/encode
-	$(CP_XPUB) gunbatch $D$(PATHBIN)/rnews.libexec/gunbatch
+	$(CP_XPUB) bunbatch $D$(PATHRNEWS)/bunbatch
+	$(CP_XPUB) c7unbatch $D$(PATHRNEWS)/c7unbatch
+	$(LI_XPUB) decode $D$(PATHRNEWS)/decode
+	$(LI_XPUB) encode $D$(PATHRNEWS)/encode
+	$(CP_XPUB) gunbatch $D$(PATHRNEWS)/gunbatch
 
 bootstrap: $(MAN)
 
