--- inn-2.3.1/innd/inndstart.c.orig	Fri Jan 26 23:29:31 2001
+++ inn-2.3.1/innd/inndstart.c	Fri Jan 26 23:32:11 2001
@@ -360,7 +360,7 @@
 	    exit(1);
 	}
     } 
-    if (!in6any)
+    if (!innconf->listenonipv6 || !in6any)
 #endif	/* INET6 */
     {
     s = socket(AF_INET, SOCK_STREAM, 0);
@@ -408,7 +408,7 @@
     innd_argv[i++] = cpcatpath(innconf->pathbin, "innd");
     innd_argv[i] = 0;
 #ifdef INET6
-    if (innconf->listeronipv6)
+    if (innconf->listenonipv6)
         printf("When starting innd, use -d -p%d [-p%d]\n", s6, s);
     else
 #endif	/* INET6 */
