--- ./nnrpd/nnrpd.c.org	2008-06-29 19:56:57.000000000 +0200
+++ ./nnrpd/nnrpd.c	2008-08-18 17:09:30.512424093 +0200
@@ -858,6 +858,12 @@ main(int argc, char *argv[])
     }
     message_program_name = xstrdup(name);
     openlog(message_program_name, L_OPENLOG_FLAGS | LOG_PID, LOG_INN_PROG);
+    /* Don't log debug unless TRACE is defined.
+     * Too much noise in /dev/log causes big load on syslog daemon.
+     */
+#ifndef TRACE
+    setlogmask(LOG_UPTO(L_NOTICE));
+#endif
     message_handlers_die(1, message_log_syslog_crit);
     message_handlers_warn(1, message_log_syslog_warning);
     message_handlers_notice(1, message_log_syslog_notice);
