--- inn-2.4.0/configure.in.orig	Tue Sep 11 21:44:00 2001
+++ inn-2.4.0/configure.in	Tue Sep 11 21:56:49 2001
@@ -45,8 +45,8 @@
 
 dnl Linking against in-tree libraries need to know the current directory (the
 dnl top of the build directory, not the source directory).
-builddir=`pwd`
-AC_SUBST(builddir)
+abs_builddir=`pwd`
+AC_SUBST(abs_builddir)
 
 dnl Earlier versions of INN used --with-largefiles, which was the wrong flag
 dnl from the perspective of what --with and --enable are supposed to mean.
@@ -1206,8 +1206,8 @@
 else
     ifelse([$5], , :, [$5])
 fi
-undefine([AC_TYPE_NAME])dnl
-undefine([AC_CV_NAME])dnl
+ifdef([AC_TYPE_NAME],[undefine([AC_TYPE_NAME])])dnl
+ifdef([AC_TYPE_NAME],[undefine([AC_CV_NAME])])dnl
 ])
 
 dnl Find a 32 bit type, by trying likely candidates.  First, check for the C9X
--- inn-2.4.0/Makefile.global.in.orig	2003-06-29 13:29:30.000000000 +0200
+++ inn-2.4.0/Makefile.global.in	2003-06-29 13:38:39.000000000 +0200
@@ -33,7 +33,7 @@
 ##  libraries built as part of INN.  Using relative paths confuses libtool
 ##  when linking the test suite.
 
-builddir	= @builddir@
+builddir	= @abs_builddir@
 
 ##  Basic compiler settings.  COPT is the variable to override on the make
 ##  command line to change the optimization or add warning flags (such as
