diff -dur inn-2.4.1.orig/Makefile.global.in inn-2.4.1/Makefile.global.in
--- inn-2.4.1.orig/Makefile.global.in	2004-02-06 12:17:58.000000000 +0100
+++ inn-2.4.1/Makefile.global.in	2004-02-06 12:18:29.748797080 +0100
@@ -251,16 +251,16 @@
 ##  article files.
 
 LI_SPRI         = $(LIBTOOL) $(INSTALL) -m 4550
-LI_XPRI         = $(LIBTOOL) $(INSTALL) -m 0550
-LI_XPUB         = $(LIBTOOL) $(INSTALL) -m 0555
+LI_XPRI         = $(LIBTOOL) $(INSTALL) -m 0750
+LI_XPUB         = $(LIBTOOL) $(INSTALL) -m 0755
 
 LI_INEWS        = $(LIBTOOL) $(INSTALL)
 LI_RNEWS        = $(LIBTOOL) $(INSTALL)
 
 CP_RPRI         = $(INSTALL) -m 0640
 CP_RPUB         = $(INSTALL) -m 0644
-CP_XPRI         = $(INSTALL) -m 0550
-CP_XPUB         = $(INSTALL) -m 0555
+CP_XPRI         = $(INSTALL) -m 0750
+CP_XPUB         = $(INSTALL) -m 0755
 
 CP_DATA	        = $(INSTALL) $(OWNER) -m $(FILEMODE) -B .OLD
 
Only in inn-2.4.1: Makefile.global.in~
