--- inn-2.5.3/Makefile.global.in.orig	2013-06-11 18:02:09.360682066 +0200
+++ inn-2.5.3/Makefile.global.in	2013-06-11 18:04:16.297346087 +0200
@@ -305,7 +305,7 @@
 
 BACKUP_OPTION	= -B .OLD
 
-LI_SPRI		= $(LIBTOOLINST) $(INSTALL) -o root -g $(RUNASGROUP) -m 4550 $(BACKUP_OPTION)
+LI_SPRI		= $(LIBTOOLINST) $(INSTALL) -m 4550 $(BACKUP_OPTION)
 LI_XPRI		= $(LIBTOOLINST) $(INSTALL) $(OWNER) -m 0550 $(BACKUP_OPTION)
 LI_XPUB		= $(LIBTOOLINST) $(INSTALL) $(OWNER) -m 0555 $(BACKUP_OPTION)
 
--- inn-2.5.3/site/Makefile.orig	2013-06-11 18:02:09.357348733 +0200
+++ inn-2.5.3/site/Makefile	2013-06-11 18:05:13.754011556 +0200
@@ -145,13 +145,9 @@
 $D$(PATH_NEWSGROUPS):	; $(CP_DATA) newsgroups.minimal $@
 $D$(PATH_ACTIVE_TIMES):
 	touch $@
-	chown $(RUNASUSER) $@
-	chgrp $(RUNASGROUP) $@
 	chmod $(FILEMODE) $@
 $D$(PATH_HISTORY):
 	touch $@
-	chown $(RUNASUSER) $@
-	chgrp $(RUNASGROUP) $@
 	chmod $(FILEMODE) $@
-	test -z "$D" && $(PATHBIN)/makedbz -i -o
+	test -z "$D" && $(PATHBIN)/makedbz -i -o || :
 
