--- inn-2.6.3/samples/cycbuff.conf.orig	2019-02-07 16:02:22.000000000 +0100
+++ inn-2.6.3/samples/cycbuff.conf	2020-11-07 18:34:28.741795625 +0100
@@ -28,9 +28,9 @@
 ##
 ##  If you're trying to stay under 2 GB, keep your sizes below 2097152.
 
-cycbuff:ONE:/export/cycbuffs/one:512000
-cycbuff:TWO:/export/cycbuffs/two:512000
-cycbuff:THREE:/export/cycbuffs/three:512000
+cycbuff:ONE:/var/spool/news/cycbuffs/one:512000
+cycbuff:TWO:/var/spool/news/cycbuffs/two:512000
+cycbuff:THREE:/var/spool/news/cycbuffs/three:512000
 
 ##  2. Meta-cyclic buffers
 ##  Format:
--- inn-2.7.2/samples/inn.conf.in.orig	2024-06-22 13:22:42.000000000 +0200
+++ inn-2.7.2/samples/inn.conf.in	2024-06-29 20:02:26.912898263 +0200
@@ -17,11 +17,11 @@
 # See the description of the mta parameter in inn.conf(5) for useful options
 # that may be worthwhile adding to the default sendmail command.
 
-mta:                         "@SENDMAIL@ -oi -oem %s"
+mta:                         "@SENDMAIL@ -oi %s"
 organization:                "A poorly-installed InterNetNews site"
 ovmethod:                    tradindexed
 hismethod:                   hisv6
-pathhost:                    @HOSTNAME@
+pathhost:                    localhost
 pathnews:                    @prefix@
 
 #runasuser:
@@ -38,8 +38,8 @@ mailcmd:                     @bindir@/in
 # Feed Configuration
 
 artcutoff:                   10
-#bindaddress:
-#bindaddress6:
+#bindaddress:                 all
+#bindaddress6:                all
 docancels:                   "require-auth"
 dontrejectfiltered:          false
 hiscachesize:                256
@@ -50,11 +50,11 @@ maxartsize:                  1000000
 maxconnections:              50
 #pathalias:
 #pathcluster:
-pgpverify:                   @DO_PGPVERIFY@
+pgpverify:                   true
 port:                        119
 remembertrash:               true
-#sourceaddress:
-#sourceaddress6:
+#sourceaddress:               any
+#sourceaddress6:              any
 verifygroups:                false
 wanttrash:                   false
 wipcheck:                    5
@@ -72,6 +72,7 @@ nfswriter:                   false
 overcachesize:               128
 ovflushcount:                50
 #ovgrouppat:
+# note: must not be changed to false if using tradspool
 storeonxref:                 true
 useoverchan:                 false
 wireformat:                  true
@@ -79,7 +80,7 @@ xrefslave:                   false
 
 # Reading
 
-allownewnews:                true
+allownewnews:                false
 articlemmap:                 true
 clienttimeout:               1800
 initialtimeout:              10
@@ -111,8 +112,8 @@ addinjectiondate:            true
 addinjectionpostingaccount:  false
 addinjectionpostinghost:     true
 checkincludedtext:           false
-#complaints:
-#fromhost:
+complaints:                  abuse@localhost
+#fromhost:                    ""
 localmaxartsize:             1000000
 #moderatormailer:
 nnrpdauthsender:             false
@@ -124,7 +125,7 @@ strippostcc:                 false
 # Posting -- Exponential Backoff
 
 backoffauth:                 false
-#backoffdb:
+#backoffdb:                   /var/lib/news/backoff
 backoffk:                    1
 backoffpostfast:             0
 backoffpostslow:             1
@@ -161,7 +162,7 @@ innwatchspoolspace:          25000
 
 # Logging
 
-docnfsstat:                  false
+docnfsstat:                  true
 htmlstatus:                  true
 incominglogfrequency:        200
 logartsize:                  true
@@ -189,7 +190,7 @@ keepmmappedthreshold:        1024
 #maxcmdreadsize:
 maxforks:                    10
 maxlisten:                   128
-nicekids:                    4
+nicekids:                    0
 nicenewnews:                 0
 nicennrpd:                   0
 pauseretrytime:              300
--- inn-2.7.0/samples/newsfeeds.in.orig	2022-07-10 15:00:35.000000000 +0200
+++ inn-2.7.0/samples/newsfeeds.in	2022-10-10 21:10:17.395989308 +0200
@@ -38,10 +38,20 @@
 ##  For the ME line (and the ME line *only*), patterns affect *outgoing*
 ##  feeds and distributions affect *incoming* feeds (including local posts).
 
+## Default of  everything to everybody except for junk, control, anything
+## with "local" as the newsgroup prefix (i.e. matches "localhost.stuff") or
+## groups under foo. Articles posted to any group under alt.binaries.warez
+## will not get propagated, even if they're cross posted to something that
+## is.
+ME\
+	:*,@alt.binaries.warez.*,!junk,!control*,!local*,!foo.*\
+        /pl,world,usa,na,gnu,bionet,pubnet,u3b,eunet,vmsnet,inet,ddn,k12\
+	::
+ 
 # Empty default subscription list, reject all incoming articles (including
 # locally posted articles) with a distribution of "local", accept all others
 # (including articles that do not have a Distribution header field).
-ME:!*/!local::
+#ME:!*/!local::
 
 # The same as the above, but would reject all posts that have
 # news.example.com in the path (posts passing through that site).
@@ -144,4 +154,4 @@ controlchan!\
 # doesn't come with INN).
 #capture!\
 #    :*/foo\
-#    :Tp,H2:/usr/local/bin/capture %s
+#    :Tp,H2:/usr/bin/capture %s
--- inn-2.7.1/samples/readers.conf.orig	2023-08-20 19:52:09.017064785 +0200
+++ inn-2.7.1/samples/readers.conf	2023-08-20 20:05:52.232605043 +0200
@@ -39,7 +39,7 @@
 # examples).  This assigns the identity of <localhost> to the local machine.
 
 auth "localhost" {
-    hosts: "localhost, 127.0.0.1, ::1, stdin"
+    hosts: "localhost, 127.0.0.1, ipv6-localhost, ::1, stdin"
     default: "<localhost>"
 }
 
