diff -urNp -x '*.orig' inn-2.6.4.org/Makefile.global.in inn-2.6.4/Makefile.global.in
--- inn-2.6.4.org/Makefile.global.in	2021-01-21 22:53:07.000000000 +0100
+++ inn-2.6.4/Makefile.global.in	2021-09-02 23:41:48.986139529 +0200
@@ -284,7 +284,7 @@ PATHTMP		= $(tmpdir)
 PATHAUTH	= $(PATHBIN)/auth
 PATHAUTHRESOLV	= $(PATHAUTH)/resolv
 PATHAUTHPASSWD	= $(PATHAUTH)/passwd
-PATHRNEWS	= $(PATHBIN)/rnews.libexec
+PATHRNEWS	= @libdir@/news/rnews
 PATHARCHIVE	= $(PATHSPOOL)/archive
 PATHARTICLES	= $(PATHSPOOL)/articles
 PATHINCOMING	= $(PATHSPOOL)/incoming
@@ -316,8 +316,8 @@ INEWSMODE	= @INEWSMODE@
 RNEWSMODE	= @RNEWSMODE@
 FILEMODE	= @FILEMODE@
 
-OWNER		= -o $(RUNASUSER) -g $(RUNASGROUP)
-ROWNER		= -o $(RUNASUSER) -g $(RNEWSGROUP)
+OWNER		=
+ROWNER		=
 
 INSTALL		= $(top)/support/install-sh -p
 
diff -urNp -x '*.orig' inn-2.6.4.org/doc/man/rnews.1 inn-2.6.4/doc/man/rnews.1
--- inn-2.6.4.org/doc/man/rnews.1	2021-01-21 22:53:07.000000000 +0100
+++ inn-2.6.4/doc/man/rnews.1	2021-09-02 23:41:48.986139529 +0200
@@ -191,7 +191,7 @@ is more widely available, due to old pat
 .PP
 Otherwise, if the command is any other word and the \fB\-a\fR flag is not
 given, \fBrnews\fR will try to execute a program with that name, looking
-for it in the directory \fIpathbin\fR/rnews.libexec.  The rest of the
+for it in the directory /usr/lib*/news/rnews .  The rest of the
 batch will be fed to that program's standard input, and the standard
 output from the program will be treated as if it were the original
 input to \fBrnews\fR.  \s-1INN\s0 comes with three such standard batch processors:
@@ -210,7 +210,7 @@ It invokes \fBgzip\fR and should be used
 .IP "\fB\-a\fR" 4
 .IX Item "-a"
 By default, \fBrnews\fR looks for additional unpackers in the
-\&\fIrnews.libexec\fR sub-directory of \fIpathbin\fR (as set in \fIinn.conf\fR).
+/usr/lib*/news/rnews directory (as set in \fIinn.conf\fR).
 If \fB\-a\fR is given, no additional batch processor will be used; only
 \&\f(CW\*(C`rnews\*(C'\fR and \f(CW\*(C`cunbatch\*(C'\fR will be recognized as valid batch commands.
 .IP "\fB\-b\fR" 4
diff -urNp -x '*.orig' inn-2.6.4.org/frontends/Makefile inn-2.6.4/frontends/Makefile
--- inn-2.6.4.org/frontends/Makefile	2021-01-21 22:53:07.000000000 +0100
+++ inn-2.6.4/frontends/Makefile	2021-09-02 23:41:48.986139529 +0200
@@ -38,11 +38,11 @@ install: all
 	for F in getlist innconfval sm ; do \
 	    $(LI_XPUB) $$F $D$(PATHBIN)/$$F ; \
 	done
-	$(CP_XPUB) bunbatch $D$(PATHBIN)/rnews.libexec/bunbatch
-	$(CP_XPUB) c7unbatch $D$(PATHBIN)/rnews.libexec/c7unbatch
-	$(LI_XPUB) decode $D$(PATHBIN)/rnews.libexec/decode
-	$(LI_XPUB) encode $D$(PATHBIN)/rnews.libexec/encode
-	$(CP_XPUB) gunbatch $D$(PATHBIN)/rnews.libexec/gunbatch
+	$(CP_XPUB) bunbatch $D$(PATHRNEWS)/bunbatch
+	$(CP_XPUB) c7unbatch $D$(PATHRNEWS)/c7unbatch
+	$(LI_XPUB) decode $D$(PATHRNEWS)/decode
+	$(LI_XPUB) encode $D$(PATHRNEWS)/encode
+	$(CP_XPUB) gunbatch $D$(PATHRNEWS)/gunbatch
 
 bootstrap: $(MAN)
 
diff -urNp -x '*.orig' inn-2.6.4.org/include/inn/paths.h.in inn-2.6.4/include/inn/paths.h.in
--- inn-2.6.4.org/include/inn/paths.h.in	2021-01-21 22:53:07.000000000 +0100
+++ inn-2.6.4/include/inn/paths.h.in	2021-09-02 23:41:48.986139529 +0200
@@ -22,12 +22,12 @@
 /* Default prefix path is pathbin. */
 #define INN_PATH_NNRPD                  "nnrpd"
 #define INN_PATH_NNTPD                  "nnrpd"
-#define INN_PATH_AUTHDIR                "auth"
+#define INN_PATH_AUTHDIR                "@libdir@/news/auth"
 #define INN_PATH_AUTHDIR_GENERIC        "generic"
 #define INN_PATH_AUTHDIR_NOPASS         "resolv"
 #define INN_PATH_AUTHDIR_PASSWD         "passwd"
 #define INN_PATH_CTLINND                "ctlinnd"
-#define INN_PATH_RNEWSPROGS             "rnews.libexec"
+#define INN_PATH_RNEWSPROGS             "@libdir@/news/rnews"
 
 /* Default prefix path is pathfilter. */
 #define INN_PATH_PERL_STARTUP_INND      "startup_innd.pl"
diff -urNp -x '*.orig' inn-2.6.4.org/scripts/inncheck.in inn-2.6.4/scripts/inncheck.in
--- inn-2.6.4.org/scripts/inncheck.in	2021-01-21 22:53:07.000000000 +0100
+++ inn-2.6.4/scripts/inncheck.in	2021-09-02 23:41:48.986139529 +0200
@@ -44,7 +44,7 @@ my %paths = (
     'passwd.nntp',	"$INN::Config::pathetc/passwd.nntp",
     'readers.conf',	"$INN::Config::pathetc/readers.conf",
     'rnews',		"$INN::Config::rnews",
-    'rnewsprogs',	"$INN::Config::pathbin/rnews.libexec",
+    'rnewsprogs',	"@libdir@/news/rnews",
     'spooltemp',	"$INN::Config::pathtmp",
     'spool',		"$INN::Config::patharticles",
     'spoolnews',	"$INN::Config::pathincoming",
diff -urNp -x '*.orig' inn-2.6.4.org/scripts/scanlogs.in inn-2.6.4/scripts/scanlogs.in
--- inn-2.6.4.org/scripts/scanlogs.in	2021-01-21 22:53:07.000000000 +0100
+++ inn-2.6.4/scripts/scanlogs.in	2021-09-02 23:41:48.986139529 +0200
@@ -10,7 +10,7 @@
 ##	norotate	Do not rotate logfiles.
 
 ##  Directory where old log files are kept.
-OLD=${MOST_LOGS}/OLD
+OLD=/var/log/archive/news
 
 ##  Files defined in innshellvars.  We repeat them for clarity.
 ERRLOG=${MOST_LOGS}/errlog
