--- inn-2.5.3/samples/innreport.conf.in.orig	2012-06-15 20:25:36.000000000 +0200
+++ inn-2.5.3/samples/innreport.conf.in	2013-06-14 19:59:36.165132651 +0200
@@ -10,7 +10,7 @@
 
 # Default parameters
 section default {
-	libpath         "@libdir@";
+	libpath         "@innlibdir@";
 	logpath		"@LOGDIR@";
 	unknown		true;        # want unknown entries.
         html-unknown    true;        # want unknown entries in HTML reports. 
--- inn-2.5.3/scripts/innshellvars.tcl.in.orig	2012-06-15 20:25:36.000000000 +0200
+++ inn-2.5.3/scripts/innshellvars.tcl.in	2013-06-14 20:10:44.008452064 +0200
@@ -23,7 +23,7 @@
 
 set inn_newsbin "${inn_pathbin}"
 set inn_newsetc "${inn_pathetc}"
-set inn_newslib "@libdir@"
+set inn_newslib "@innlibdir@"
 set inn_innddir "${inn_pathrun}"
 set inn_locks "${inn_pathrun}"
 set env(NEWSBIN) "$inn_newsbin"
--- inn-2.5.3/scripts/innshellvars.in.orig	2012-06-15 20:25:36.000000000 +0200
+++ inn-2.5.3/scripts/innshellvars.in	2013-06-14 20:30:54.805093495 +0200
@@ -20,7 +20,7 @@
 
 NEWSBIN=${PATHBIN}
 NEWSETC=${PATHETC}
-NEWSLIB=@libdir@
+NEWSLIB=@innlibdir@
 INNDDIR=${PATHRUN}
 LOCKS=${PATHRUN}
 export NEWSBIN NEWSETC NEWSLIB INNDDIR LOCKS
--- inn-2.5.3/scripts/innshellvars.pl.in.orig	2012-06-15 20:25:36.000000000 +0200
+++ inn-2.5.3/scripts/innshellvars.pl.in	2013-06-14 20:31:17.728426352 +0200
@@ -28,7 +28,7 @@
 
 $newsbin = $pathbin;
 $newsetc = $pathetc;
-$newslib = "@libdir@";
+$newslib = "@innlibdir@";
 $innddir = $pathrun;
 $locks = $pathrun;
 $ENV{'NEWSBIN'} = $newsbin;
--- inn-2.5.3/Makefile.global.in.orig	2013-06-14 15:36:10.805462084 +0200
+++ inn-2.5.3/Makefile.global.in	2013-06-14 21:05:53.935049745 +0200
@@ -225,6 +225,7 @@
 docdir		= @docdir@
 includedir	= @includedir@
 libdir		= @libdir@
+datadir		= @datadir@
 mandir		= @mandir@
 sysconfdir	= @sysconfdir@
 tmpdir		= @tmpdir@
@@ -237,6 +238,7 @@
 PATHMAN		= $(mandir)
 PATHINCLUDE	= $(includedir)
 PATHLIB		= $(libdir)
+PATHINNLIB	= @innlibdir@
 PATHLIBPERL	= @LIBPERLDIR@
 PATHCONTROL	= @CONTROLDIR@
 PATHFILTER	= @FILTERDIR@
--- inn-2.5.3/perl/INN/Config.pm.in.orig	2012-06-15 20:25:36.000000000 +0200
+++ inn-2.5.3/perl/INN/Config.pm.in	2013-06-14 21:06:40.531715437 +0200
@@ -37,7 +37,7 @@
 $spooltemp $path);
 our $newshome = $pathnews;
 our $home = $pathnews;
-our $newslib = "@libdir@";
+our $newslib = "@innlibdir@";
 our $newsbin = $pathbin;
 our $controlprogs = $pathcontrol;
 our $newslbin = "$pathnews/local";
--- inn-2.5.3/support/fixscript.in.orig	2012-06-15 20:25:36.000000000 +0200
+++ inn-2.5.3/support/fixscript.in	2013-06-14 21:08:38.721712975 +0200
@@ -25,6 +25,7 @@
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 libdir="@libdir@"
+innlibdir="@innlibdir@"
 libperldir="@LIBPERLDIR@"
 
 # We can probably just assume sed is on the path, but since we have it, we may
@@ -60,7 +61,7 @@
 case "$interpreter" in
 */sh|*SH*)
     path="@SHELL@"
-    lib=". $libdir/innshellvars"
+    lib=". $innlibdir/innshellvars"
     ;;
 */perl*|*PERL*)
     path=`echo "$interpreter" | sed 's%^#! *[^ ][^ ]*%'"$PERLPATH%"`
--- inn-2.5.3/support/fixconfig.in.orig	2012-06-15 20:25:36.000000000 +0200
+++ inn-2.5.3/support/fixconfig.in	2013-06-14 21:09:38.871711720 +0200
@@ -17,6 +17,7 @@
 bindir=@bindir@
 libexecdir=@libexecdir@
 libdir=@libdir@
+innlibdir=@innlibdir@
 sbindir=@sbindir@
 sysconfdir=@sysconfdir@
 
@@ -59,6 +60,7 @@
      -e "s,@bindir[@],$bindir,g" \
      -e "s,@libexecdir[@],$libexecdir,g" \
      -e "s,@libdir[@],$libdir,g" \
+     -e "s,@innlibdir[@],$innlibdir,g" \
      -e "s,@sbindir[@],$sbindir,g" \
      -e "s,@sysconfdir[@],$sysconfdir,g" \
      -e "s,@CONTROLDIR[@],$CONTROLDIR,g" \
--- inn-2.5.3/m4/paths.m4.orig	2012-06-15 20:25:36.000000000 +0200
+++ inn-2.5.3/m4/paths.m4	2013-06-14 21:13:15.238373880 +0200
@@ -47,6 +47,9 @@
 _INN_ARG_DIR([tmp], ['${prefix}/tmp'], [tmpdir],
     [AS_HELP_STRING([--with-tmp-dir=PATH],
         [Path for temporary files [PREFIX/tmp]])])
+_INN_ARG_DIR([innlib], ['${datadir}/news'], [innlibdir],
+    [AS_HELP_STRING([--with-innlib-dir=PATH],
+        [Path to inn library scripts [DATADIR/news]])])
 
 dnl Some additional paths used by inn/paths.h.
 eval PATH_CONFIG="$sysconfdir"
--- inn-2.5.3/scripts/Makefile.orig	2012-06-15 20:25:36.000000000 +0200
+++ inn-2.5.3/scripts/Makefile	2013-06-14 21:14:08.268372773 +0200
@@ -27,7 +27,7 @@
 	done
 	for F in innreport_inn.pm innshellvars innshellvars.pl \
 	         innshellvars.tcl ; do \
-	    $(CP_RPUB) $$F $D$(PATHLIB)/$$F ; \
+	    $(CP_RPUB) $$F $D$(PATHINNLIB)/$$F ; \
 	done
 
 bootstrap:
